subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/impl/DefaultFileSystemLocationFingerprint.java

```diff
@@ -40,6 +40,7 @@ private static HashCode hashForType(FileType fileType, HashCode hash) {
             case Directory:
                 return DIR_SIGNATURE;
             case Missing:
+            case Unavailable:
```

Comment:
I don't think we need to fingerprint the `Unavailable` file type. I actually thought we should still fail when we encounter it.

Reply:
Shouldn't we detect changes to unaccessible files (e.g. becomes readable on next try)?
If yes, maybe we should add a test for that.
Please see:
<img width="954" alt="Screen Shot 2019-06-28 at 11 30 35" src="https://user-images.githubusercontent.com/1841944/60329031-2b2da080-9998-11e9-874d-348c416137b7.png">

Reply:
This was changed in the latest code, please check if your concerns were addressed.

Reply:
I think that fingerprinting an `Unavailable` file should cause a build failure. I think that `hashForType` should throw an exception and `UNAVAILABLE_FILE_SIGNATURE` should be removed. Would this cause any tests to fail?

Reply:
We should not detect changes to unavailable files in change detection - the build should fail when those files are used as inputs/outputs of a task.

Reply:
Can we do that without changing the output fingerprinting to use snapshotting instead?
This is the only place where I've handled `case Unavailable:` explicitly - the new test (`unreadable #type not produced by task is ignored`) needs this currently.

Reply:
Yeah, you are right. This is currently not possible since we use the fingerprints for overlapping output detection :(
We already use the snapshots when filtering the output, so that does not need to change.

Given the impact of ☝️ , let's do it in a follow up PR and allow fingerprinting unavailable files in this PR. We should disallow fingerprinting unavailable files for 5.6 though.

Reply:
k, I've added it to #9034

Reply:
This is not #9034, it is a smaller step in that direction. For #9034 we also need to store snapshots in the task history. For ☝️ , we don't need to do anything about the task history or about comparing two snapshots - which are the expensive parts..